scripts/operations/configuration/write_root_secrets_to_vault.py

```diff
@@ -76,38 +85,129 @@ def get_root_hash_from_etc_shadow() -> str:
+    """
+    logging.info(
+        f"Reading in SSH private key from '{file_name}' file")
+    return args.get_text_file_contents(
+        file_name=file_name,
+        value_validator=lambda s: args.validate_string(s, min_length=256))
+
+
+def pub_ssh_key_file(file_name: str) -> str:
+    """
+    Wrapper for args.get_text_file_contents with appropriate arguments for SSH key files.
+    At this point, we just make sure that the files are at least 256 characters long.
+    """
+    logging.info(
+        f"Reading in SSH public key from '{file_name}' file")
+    return args.get_text_file_contents(
+        file_name=file_name,
+        value_validator=lambda s: args.validate_string(s, min_length=256))
+
+
+def main():
+    """
+    Parses the command line arguments, read in the secrets, write them to Vault.
+    """
+    logging.debug(f"Command line arguments: {sys.argv}")
+    
+    parser = argparse.ArgumentParser(
+        description="Update CSM root secrets in Vault with specified SSH keys and password hash")
+
+    # Password source arguments are mutually exclusive
+    pw_group = parser.add_mutually_exclusive_group()
+    pw_group.add_argument("--pw-env-var", type=pw_env_var,
+                          metavar="PASSWORD_ENV_VAR_NAME",
+                          help="Read plaintext password from specified variable")
+    pw_group.add_argument("--pw-hash-env-var", type=pw_hash_env_var,
+                          metavar="PASSWORD_HASH_ENV_VAR_NAME",
+                          help="Read shadow password hash string from specified variable")
+    pw_group.add_argument("--pw-sys", action='store_true',
+                          help="Read password hash from /etc/shadow file on the system (default)")
+    pw_group.add_argument("--pw-prompt", action='store_true',
+                          help="Prompt user to enter plaintext password")
+
+    parser.add_argument("--pri-key-file", type=pri_ssh_key_file,
+                        metavar='private_key_file', default=PRI_KEY_PATH, dest='pri_key',
+                        help=f"Path to private key file (default: {PRI_KEY_PATH})")
+
+    parser.add_argument("--pub-key-file", type=pub_ssh_key_file,
+                        metavar='public_key_file', default=PUB_KEY_PATH, dest='pub_key',
+                        help=f"Path to public key file (default: {PUB_KEY_PATH})")
+
+    parsed_args = parser.parse_args()
+
+    # Get password hash
+    if parsed_args.pw_hash_env_var is not None:
+        # Simplest case -- we have already read in the hashed value from the environment variable
+        pw_hash = parsed_args.pw_hash_env_var
+    elif parsed_args.pw_env_var is not None:
+        # Generate the hash
+        logging.debug("Generating password hash")
+        pw_hash = crypt.crypt(parsed_args.pw_env_var)
+    elif parsed_args.pw_prompt:
+        # Prompt user for the password
+        logging.debug("Prompting user for password")
+        pw_text = getpass.getpass("Enter desired password: ")
```

Comment:
It could be marginally more user friendly to do this in a loop, e.g.:
```
passwords_match = False
while not passwords_match:
  pw_text = getpass.getpass("Enter desired password: ")
  passwords_match = pw_text == getpass.getpass("Verify password: ")
  if not passwords_match:
    logging.error("Passwords do not match. Please try again.")
```

Also definitely not necessary but might be nice to refactor this prompting into a helper function.

Reply:
Good call. Done!

Reply:
Refactored version looks good to me too.